lib/puppet/parser/resource.rb

```diff
@@ -184,6 +184,15 @@ def name
   # if we ever receive a parameter named 'tag', set
   # the resource tags with its value.
   def set_parameter(param, value = nil)
+    if evaluated?
+      strict = Puppet[:strict]
+      if :off != strict
```

Comment:
One could argue that this is not a matter of style. Indentation, ways to comment, whether or not to use `and` or `or`, etc. is style. This is efficient programming. I changed it nevertheless, since it should be safe to assume that `Puppet[:strict]` contains a symbol.